lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -714,8 +714,58 @@ def _post(
             Response to return to client
         """
 
+        return self.common_ES(params, request, context)
+
+    def _delete(
+        self, params: ApiParams, request: Request, context: ApiContext
+    ) -> Response:
+        """
+        Perform the requested DELETE operation, and handle any exceptions.
+
+        This is called by the ApiBase delete() method through its dispatch
+        method, which provides parameter validation.
+
+        NOTE: This method relies on a ParamType.DATASET parameter being part of
+        the DELETE API Schema defined by any subclass that extends this base
+        class, and the DELETE schema must select DATASET authorization.
+        (This is checked by the constructor.)
+
+        Args:
+            params: Type-normalized client parameters
+            request: Original incoming Request object (not used)
+            context: API context
+
+        Returns:
+            Response to return to client
+        """
+        return self.common_ES(params, request, context)
+
+    def common_ES(
+        self, params: ApiParams, request: Request, context: ApiContext
+    ) -> Response:
+        """
+        Perform the requested operation, and handle any exceptions.
+
+        Args:
+            params: Type-normalized client parameters
+            request: Original incoming Request object (not used)
+            context: API context
+
+        Returns:
+            Response to return to client
+        """
+
+        if context["attributes"]:
+            if context["attributes"].require_stable:
+                self.require_stable = context["attributes"].require_stable
+            if context["attributes"].require_map:
+                self.require_map = context["attributes"].require_map
+            if context["attributes"].action:
+                self.action = context["attributes"].action
```

Comment:
I think you should avoid this and simply reference `context["attributes"]` directly, for example at line 781, `if context["attributes"].require_stable and operation`.

I think that any `ElasticBulkBase` API that can get here will have `context["attributes"]` ... which is somewhat illustrated already by this code since you're leaving `self.require_stable` *undefined* by default, which would cause an exception below. You could enforce that in the constructor, "to be safe".